# **MÓDULO 13**
# Projeto - Fundamentos da Descoberta de Dados

Nesse projeto trabalharemos com a base de dados de produtos de um supermercado do Chile.
A ideia é que vocês apliquem os conceitos estatísticos vistos no último módulo, mais os conceitos de visualizações de dados através de gráficos e finalizem publicando no seu github!

In [2]:
import pandas as pd
import matplotlib.pyplot as plt
import plotly.express as px

Faça a leitura dos dados do arquivo CSV:


Altere o código abaixo de acordo com seu diretório.

In [35]:
df = pd.read_csv("./MODULO7_PROJETOFINAL_BASE_SUPERMERCADO - MODULO7_PROJETOFINAL_BASE_SUPERMERCADO .csv", delimiter=',')
print(df.dtypes)
df.head(10)

title             object
Marca             object
Preco_Normal       int64
Preco_Desconto     int64
Preco_Anterior     int64
Desconto           int64
Categoria         object
dtype: object


title       Marca  Preco_Normal  \
0              Pack 12 un, Leche extra proteína 1 L  Loncoleche         19788   
1   Pack 12 un, Leche chocolate receta original 1 L     Soprole         18228   
2    Pack 12 un, Leche semidescremada chocolate 1 L     Soprole         18228   
3     Pack 12 un, Leche semidescremada frutilla 1 L     Soprole         18228   
4       Pack 12 un, Leche sin lactosa chocolate 1 L  Loncoleche         17988   
5        Pack 12 un, Leche sin lactosa frutilla 1 L  Loncoleche         17988   
6  Pack 12 un, Leche saborizada light chocolate 1 L  Loncoleche         17988   
7         Pack 12 un, Leche saborizada frutilla 1 L       Colun         17388   
8         Pack 12 un, Leche saborizada vainilla 1 L       Colun         17388   
9           Pack 12 un, Leche saborizada manjar 1 L       Colun         17388   

   Preco_Desconto  Preco_Anterior  Desconto Categoria  
0               0               0         0   lacteos  
1               0               0         0   lacteos  
2               0               0         0   lacteos  
3               0               0         0   lacteos  
4               0               0         0   lacteos  
5               0               0         0   lacteos  
6               0               0         0   lacteos  
7               0               0         0   lacteos  
8               0               0         0   lacteos  
9               0               0         0   lacteos

Os campos do nosso dataframe são:

**Title:** Nome do produto.


**Marca:** A marca do produto.


**Preco_Normal:** O preço em que o produto costuma ser vendido quando não há desconto.


**Preco_Desconto:** O preço vendido após o desconto ser aplicado.


**Preco_Anterior:** Preço em que era comercializado o produto antes do desconto aplicado.


**Desconto:** Total de desconto aplicado.






As colunas que aparecem com valores 0 são para os produtos onde não tivemos descontos aplicados.


As categorias estão em espanhol!

# 1 - Traga a média e a mediana dos preços - coluna Preco_Normal - por categoria de produto.
# Identifique as categorias que parecem ter um valor de média abaixo ou acima da mediana.

In [21]:
df.groupby('Categoria')['Preco_Normal'].mean()

Categoria
belleza-y-cuidado-personal    1783.556485
comidas-preparadas            3095.043478
congelados                    2108.042553
frutas                        1724.473684
instantaneos-y-sopas           765.491228
lacteos                       2385.219239
verduras                      1343.296875
Name: Preco_Normal, dtype: float64

In [22]:
df.groupby('Categoria')['Preco_Normal'].median()

Categoria
belleza-y-cuidado-personal    1569.0
comidas-preparadas            3290.0
congelados                    1519.0
frutas                        1195.0
instantaneos-y-sopas           439.0
lacteos                        989.0
verduras                      1180.0
Name: Preco_Normal, dtype: float64

Ao comparar a tabela gerada para a média e mediana, podemos perceber que Belleza-y-cuidado-personal, congelados, frutas, instantaneos-y-sopas, lacteos e verduras possuem um valor de média acima do valor de mediana, enquanto comidas-preparadas apresenta um valor de média inferior ao valor da mediana.

# 2 - Traga o desvio padrão por categoria de produto.
# Qual o comportamento da média e mediana nas categorias com maior desvio?

In [23]:
df.groupby('Categoria')['Preco_Normal'].std()

Categoria
belleza-y-cuidado-personal    2210.041719
comidas-preparadas            2019.911428
congelados                    2111.539896
frutas                        1639.151114
instantaneos-y-sopas          1170.232869
lacteos                       3925.816164
verduras                      1012.699625
Name: Preco_Normal, dtype: float64

As categorias com maior desvio são lacteos, beleza y cuidado personal, congelados e comidas-preparadas. No caso dos produtos Lácteos que apresentaram maior desvio padrão, temos uma média de 2385 e uma mediana de 989, indicando uma disparidade muito grande entre as duas, o que indica que a maioria dos produtos é relativamente barata, devido ao valor de mediana mais baixa, porém existem produtos mais caros que fazem com que a média apresente valor consideravelmente mais alto que a mediana. Quando analisamos as outras categorias com desvio mais alto como beleza y cuidado personal, congelados e comidas-preparadas, percebemos que a média e mediana estão mais próximas entre si, mas ainda existe um efeito moderado de distorção da média com relação ao valor central dos preços.
Contudo, quando analisamos categorias com menor desvio, como verduras, frutas e instantaneos e sopas, percebemos que os valores da média e mediana estão mais próximos entre si, o que indica que existe uma concentração maior de preços e a média sofre menor influencia de outliers.

# 3 - Plot um boxplot da distribuição do Preco_Normal para a categoria que você identificou que tem o maior desvio padrão. Como é a distribuição desses dados segundo o boxplot? Você identifica muitos outliers?

Dica: Para trazer apenas os dados da categoria que você deseja você pode usar o df.loc[df['Categoria'] == 'CATEGORIA ESCOLHIDA'

In [28]:
px.box(df.loc[df['Categoria'] == 'lacteos'], x='Preco_Normal',title='Boxplot dos produtos lacteos')

O boxplot apresentou um comportamento condizente com o observado na comparação entre a média e mediana, onde os produtos lácteos possuem um número significativo de outliers que acabam distorcendo a média e deslocando-a oara valores superiores aos valores típicos da maioria dos produtos.

# 4 - Plote um gráfico de barras onde temos a média de descontos por categoria.

In [30]:
media_desconto = df.groupby('Categoria')['Desconto'].mean()

fig = px.bar(media_desconto, x=media_desconto.index, y=media_desconto.values, title='Média de Desconto por Categoria')
fig.show()

# 5 - Plote um gráfico de mapa interativo agrupando os dados por categoria, marca e trazendo a média de desconto.

In [34]:
df_map = df.groupby(['Categoria', 'Marca'],as_index=False)['Desconto'].mean().rename(columns={'Desconto':'Desconto_Medio'})
fig=px.treemap(df_map, path=['Categoria', 'Marca'], values='Desconto_Medio', title='Média de Desconto por Categoria e Marca')
fig.update_traces(hovertemplate="<b>%{label}</b><br>Desconto médio: %{value:.2f}<extra></extra>")
fig.show()